lib/confluence/failure_to_ship.es6.js

```diff
@@ -41,92 +41,78 @@ foam.CLASS({
-              this.AND(
-                  this.GT(this.Release.RELEASE_DATE, previousYear),
-                  this.LT(this.Release.RELEASE_DATE, date))
-              )).select().then((result) => {
-            // releaseIface are a map of interfaces from this release in
-            // recent one year's release.
-            let releaseIface = {};
-            let numOtherReleases = 0;
-            // majorIface are a map of interfaces of form:
-            //     {interfaceKey: <number of appearances>}.
-            // Interfaces with number of appearances equals to numOtherReleases
-            // are considered as major interfaces.
-            let majorIface = {};
-            let promises = [];
-            let prevReleases = [];
-            let comparedReleases = [];
-            for (let i = 0; i < result.a.length; i++) {
-              if (result.a[i].browserName !== release.browserName) {
-                numOtherReleases++;
-                comparedReleases.push(result.a[i]);
-              } else if (result.a[i].releaseKey !== release.releaseKey) {
-                prevReleases.push(result.a[i]);
-              }
-              promises.push(result.a[i].interfaces.dao.select(
-                  this.AnonymousSink.create({sink: {put: function(iface) {
-                    if (result.a[i].browserName === release.browserName) {
-                      releaseIface[iface.interfaceKey] = true;
-                    } else {
-                      // Interfaces may have names defined in majorIface's
-                      // prototype chain; that's why we don't use
-                      // majorIface[iface.interfaceKey] =
-                      //     (majorIface[iface.interfaceKey] || 0) + 1.
-                      if (!majorIface.hasOwnProperty(iface.interfaceKey)) {
-                        majorIface[iface.interfaceKey] = 0;
-                      }
-                      majorIface[iface.interfaceKey]++;
-                    }
-                  }}})));
-            }
-            return Promise.all(promises).then(() => {
-              // Find interfaces that all other browsers' releases supports
-              // but not available in any version of this browser's releases.
-              let numFailureToShip = 0;
-              for (let ifaceKey in majorIface) {
-                if (!majorIface.hasOwnProperty(ifaceKey)) continue;
-                if (majorIface[ifaceKey] !== numOtherReleases) continue;
-                if (!releaseIface.hasOwnProperty(ifaceKey)) {
-                  numFailureToShip++;
-                }
-              }
-              return this.failureToShipDAO.put(this.FailureToShipData.create(
-                {
+        return Promise.all(releases.map(
+            release => this.computeForRelease(releases, date, release)));
+      },
+    },
+    {
+      name: 'computeForRelease',
+      documentation: 'Compute failure to ship value for a single release.',
+      code: function(releases, date, release) {
```

Comment:
This method looks very similar to the one I just reviewed. Maybe it should be shared?